src/app-data/state/producers/measurements.ts

```diff
@@ -108,6 +108,15 @@ export const selectMeasurementKind: AppStateProducer<
 export const setMeasurementVisibility: AppStateProducer<
   'SET_MEASUREMENT_VISIBILITY'
 > = (draft, action) => {
+  if (action.payload.id === null) {
+    for (const id of Object.keys(draft.view.measurements)) {
```

Comment:
You should only update the selected kind, like with "select all"

Reply:
do i have to update, only the selected items ?

Reply:
yes

Reply:
okay, so i will mark this conversation as resolved :)

Reply:
Actually, we discussed last week and the issue was wrong (I just updated it)
There should be two separate buttons with the eyes (to show and hide). Click on the buttons should only apply to the selected measurements.

For example, with this selection:
![image](https://user-images.githubusercontent.com/2352663/204302895-a11440c5-cd49-4f3a-8168-503f2803a4e1.png)

After the click to the open eye, it should be like this:
![image](https://user-images.githubusercontent.com/2352663/204303576-f2aa24a5-0e46-4823-bc6c-8ee1f47f29d5.png)